src/internal/bolt/bolt-protocol-v1.js

```diff
@@ -302,4 +313,68 @@ export default class BoltProtocol {
   _createUnpacker (disableLosslessIntegers) {
     return new v1.Unpacker(disableLosslessIntegers)
   }
+
+  /**
+   * Write a message to the network channel.
+   * @param {RequestMessage} message the message to write.
+   * @param {ResultStreamObserver} observer the response observer.
+   * @param {boolean} flush `true` if flush should happen after the message is written to the buffer.
+   */
+  write (message, observer, flush) {
+    const queued = this._queueObserver(observer)
```

Comment:
Maybe the write function would be clearer if you expand the _queueObserver function in this function. Handling of the _fatalError is a bit obscured by something that looks optional. Or at least rename _queueObserver...

Reply:
I will make this more evident.